.github/workflows/build.yml

```diff
@@ -0,0 +1,41 @@
+name: build
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  release:
+    types:
+      - released
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+permissions: read-all
+
+jobs:
+  cmake:
+    strategy:
+      matrix:
+        runs-on:
+          - ubuntu-latest
+          # macos-13 is an intel runner, macos-14 is apple silicon
+          - macos-13
+          - macos-latest
+      fail-fast: false
+    runs-on: ${{ matrix.runs-on }}
+    timeout-minutes: 30
+    steps:
+      - uses: zacharyburnett/setup-abseil-cpp@de39f445295c887839e30c864ffbbb1c0231bc83 # 1.0.5
```

Comment:
Just wondering, what's the difference between `setup-abseil-cpp` and using `vcpkg` or `homebrew` to install `abseil-cpp`?  I see both `vcpkg` and `homebrew` are installed on the [runner images](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2404-Readme.md).

Reply:
This was a while ago, but I think I remember that the `homebrew` recipe was not configured such that it could link properly, which is why I resorted to building it manually; I did not try `vcpkg` though